### ISSUE TITLE
REST API ignores master password

### ISSUE THREAD

**Issue by Quinity** [fcturner/passwords#248]
When using the REST API of your password app, it bypasses the master password and accesses the stored password by using the basic authentication only.

**Comment by fcturner** [comment]
Yes, the master password is web only for now. Authentication with a RESful API (i.e. connect to my Passwords app) like ownCloud created it, is only possible with a username and one password. So, I don't know of a way to implement the master password on remote access too.

Maybe later!

**Comment by fungs** [comment]
I was gonna post the same issue. Setting a master password I expected to create a (second) token for password access, as most software does. However, the master password is more or less useless if it can be bypassed via API. So for me, the only real use of a custom master password is to use it for encrypting the actual data (instead of using a log-password derived token). So there is an access token, the oc/nc login, and an encryption/decryption token, the master password. Lazy users should use an encryption/decryption token which is derived from their login. This way, privacy-aware users will have both layers of protection.

IMO it is really bad practice to derive cryptographic tokens from web-login passwords due to phishing, javascript etc.
